### PR TITLE
CXX-2400 Add support for the comment field to all helpers

### PR DESCRIPTION
**Pull request by eramongodb**
## Description

This PR resolves CXX-2400, CXX-2447, CXX-2450, CXX-2463, CXX-2475, CXX-2486, CXX-2498, and CXX-2519.

Only CRUD unified test files are updated. Updates for change streams unified test files are deferred to CXX-2564.

## Add support for expectResult as array of root documents

The unified test runner needed to be updated to allow for [extra fields in `expectResult` for array elements](https://github.com/mongodb/specifications/blob/b63ea83e782b1fe2081ff0ad38bd088525396f41/source/unified-test-format/unified-test-format.rst#allowing-extra-fields-in-root-level-documents) of certain test operations. The operations that return "an array of root documents" were identified by examining conditions when [the C driver assigns sets `array_of_root_docs` to true](https://github.com/mongodb/mongo-c-driver/blob/f35c380df69f97e6f6a739707cb9917ad033f900/src/libmongoc/tests/unified/result.c#L36):
- aggregate
- find
- iterateUntilDocumentOrError
- listCollections
- listDatabases
- listIndexes

### CODE REVIEW

**Review by rcsanchez97 — APPROVED** [review]
LGTM

File: src/mongocxx/options/aggregate.hpp [hunk 1]
```diff
@@ -276,6 +277,31 @@ class MONGOCXX_API aggregate {
     ///
     const stdx::optional<class read_concern>& read_concern() const;
 
+    ///
+    /// Sets the comment to use for this operation.
+    ///
+    /// @see
+    ///   https://docs.mongodb.com/manual/reference/command/aggregate/
+    ///
+    /// @param read_concern
```
Comment (kevinAlbs) [inline src/mongocxx/options/aggregate.hpp:286]:
```suggestion
    /// @param comment
```

File: src/mongocxx/options/delete.cpp [hunk 2]
```diff
@@ -57,6 +57,15 @@ const stdx::optional<bsoncxx::document::view_or_value> delete_options::let() con
     return _let;
 }
 
+delete_options& delete_options::comment(bsoncxx::document::element comment) {
```
Comment (kevinAlbs) [inline src/mongocxx/options/delete.cpp:60]:
```suggestion
delete_options& delete_options::comment(bsoncxx::types::bson_value::view_or_value comment) {
```

**Review by kevinAlbs — COMMENTED** [review]
Also consider:
- Add `comment` to `options::distinct` and sync specification tests added in CXX-2519.
- Add `comment` to `options::estimated_document_count` and `options::count` and sync specification tests added in CXX-2463.

**Comment by eramongodb** [comment]
Synced unified CRUD tests up to CXX-2519 as requested, including an intermediate ticket CXX-2498.

**Comment by eramongodb** [comment]
VS 2017 build issues on Evergreen will be addressed separately in https://github.com/mongodb/mongo-cxx-driver/pull/882.

**Pull request merged**